flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/PulsarSinkBuilder.java

```diff
@@ -315,6 +315,11 @@
         if (!configBuilder.contains(PULSAR_PRODUCER_NAME)) {
             LOG.warn(
                     "We recommend set a readable producer name through setProducerName(String) in production mode.");
+        } else {
+            String producerName = configBuilder.get(PULSAR_PRODUCER_NAME);
+            if (!producerName.contains("%s")) {
+                configBuilder.override(PULSAR_PRODUCER_NAME, producerName + " - %s");
```

Comment:
What is the `%s` supposed to do? I haven't found anything in the Pulsar docs. Could you add a comment to clarify?

Reply:
We will replace the %s as a UUID in [PulsarSinkConfigUtils.java](https://github.com/apache/flink/pull/19285/files#diff-a43e1010653a205454f362b6a20b92a5ce2bde863b47fcc851b9eb37a9d404fe). This would make sure all the producer name unique. Same in `PulsarSourceBuilder`.

Reply:
Ah I see. Thank you, I missed that bit.